src/transformers/models/owlv2/convert_owlv2_to_hf.py

```diff
@@ -0,0 +1,331 @@
+        rename_keys.append(("objectness_head/Dense_1/kernel", "objectness_head.dense1.weight"))
+        rename_keys.append(("objectness_head/Dense_1/bias", "objectness_head.dense1.bias"))
+        rename_keys.append(("objectness_head/Dense_2/kernel", "objectness_head.dense2.weight"))
+        rename_keys.append(("objectness_head/Dense_2/bias", "objectness_head.dense2.bias"))
+
+    # fmt: on
+
+    return rename_keys
+
+
+def rename_and_reshape_key(dct, old, new, config):
+    val = dct.pop(old)
+
+    if ("out_proj" in new or "v_proj" in new or "k_proj" in new or "q_proj" in new) and "vision" in new:
+        val = val.reshape(-1, config.vision_config.hidden_size)
+    if ("out_proj" in new or "v_proj" in new or "k_proj" in new or "q_proj" in new) and "text" in new:
+        val = val.reshape(-1, config.text_config.hidden_size)
+
+    if "patch_embedding" in new:
+        print("Reshaping patch embedding... for", new)
+        val = val.transpose(3, 2, 0, 1)
+    elif new.endswith("weight") and "position_embedding" not in new and "token_embedding" not in new:
+        val = val.T
+
+    if new.endswith("bias"):
+        val = val.reshape(-1)
+
+    dct[new] = torch.from_numpy(np.array(val))
+
+
+@torch.no_grad()
+def convert_owlv2_checkpoint(model_name, pytorch_dump_folder_path, push_to_hub):
+    """
+    Copy/paste/tweak model's weights to our OWL-ViT structure.
+    """
+    config = get_owlv2_config(model_name)
+
+    # Load original state dict based on model name
+    model_name_to_checkpoint_path = {
+        "owlvit-base-patch16": "/Users/nielsrogge/Documents/OWL-ViT/clip_vit_b16_6171dab",
+        "owlv2-base-patch16": "/Users/nielsrogge/Documents/OWL-ViT/owl2-b16-960-st-ngrams_c7e1b9a",
+        "owlv2-base-patch16-ensemble": "/Users/nielsrogge/Documents/OWL-ViT/owl2-b16-960-st-ngrams-curated-ft-lvisbase-ens-cold-weight-05_209b65b",
+    }
+    checkpoint_path = model_name_to_checkpoint_path[model_name]
+    variables = checkpoints.restore_checkpoint(checkpoint_path, target=None)
+    variables = variables["params"] if "v2" in model_name else variables["optimizer"]["target"]
+    flax_params = jax.tree_util.tree_map(lambda x: x.astype(jnp.float32) if x.dtype == jnp.bfloat16 else x, variables)
+    state_dict = flatten_nested_dict(flax_params)
+
+    for name, param in state_dict.items():
+        print(name, param.shape, param.dtype)
+
+    # Rename keys
+    rename_keys = create_rename_keys(config, model_name)
+    for src, dest in rename_keys:
+        rename_and_reshape_key(state_dict, src, dest, config)
+
+    # load HuggingFace model
+    model = Owlv2ForObjectDetection(config)
+    missing_keys, unexpected_keys = model.load_state_dict(state_dict, strict=False)
+    print("Missing keys:", missing_keys)
+    print("Unexpected keys:", unexpected_keys)
+    assert missing_keys == ["owlv2.visual_projection.weight"]
+    assert unexpected_keys == [] if "v2" in model_name else ["class_head/padding", "class_head/padding_bias"]
```

Comment:
I thin this might be confusing we should add in the proint that this is expected maybe?